### ISSUE TITLE
Migrate code implementation to kubeflow/common fashion

### ISSUE THREAD

**Issue by Jeffwan** [kubeflow/pytorch-operator#282]
kubeflow/common release a stable version 0.3.1 and we can migrate to use implementation of kubeflow/common. 

The change will be similar to change https://github.com/kubeflow/tf-operator/pull/1171. 

It would be better to resolve dependencies, upgrade to 0.3.1 for interfaces, and then update implementation. 

Hope this can be part of 1.1 release.

**Comment by johnugeorge** [comment]
I am worried if it is little late for the 1.1 release. TF-operator one too has not been tested extensively with common.

**Comment by Jeffwan** [comment]
@johnugeorge Yeah, If we can not catch 1.1 release, I think we can postpone to next release. As TF and Pytorch already graduate to v1, we definitely need more test to avoid regression issues.
